std_execution.bs

```diff
@@ -553,6 +553,293 @@ merged into a single one in an asynchronous fashion.
+invoke the default implementation which returns `std::never_stop_token`. This is a special `stoppable_token` type that
+is statically known to always return `false` from the `stop_possible()` method.
+
+Sender code that tries to use this stop-token will in general result in code that handles stop-requests being
+compiled out and having little to no run-time overhead.
+
+If the sender doesn't call `execution::get_stop_token()`, for example because the operation does not support
+cancellation, then it will simply not respond to stop-requests from the caller.
+
+Note that stop-requests are generally racy in nature as there is often a race betwen an operation completing
+naturally and the stop-request being made. If the operation has already completed or past the point at which
+it can be cancelled when the stop-request is sent then the stop-request may just be ignored. Thus an application
+will typically need to be able to cope with senders that might ignore a stop-request anyway.
+
+### Cancellation and eager operations ###
+
+TODO: Talk about limitations of using cancellation with operation that might execute eagerly.
+
+* Eager operations won't have access to the caller's stop-token before they start executing.
+  Will need to wait until the sender is connected/started before it can get access to the stop-token of the caller.
+* What happens if the destructor of an eager-sender is run before the operation is connected?
+* What happens if the destructor of an operation-state created from an eager-sender is run before `execution::start()` is called?
+* Should it be required that the return from the destructors strongly-happens after the completion of any execution of the corresponding operations launched by that sender?
+* Should it be required that the destructor issues a stop-request before blocking on completion of the operation?
+* Risks of this approach are deadlocks if the destructor is run on an execution context with weak forward progress.
+  e.g. a thread-pool with a maximum thread-count
+
+### Cancellation is inherently racy ###
+
+Usually, an operation will attach a stop-callback at some point inside the call to `execution::start()` so that
+a subsequent stop-requests will interrupt the logic.
+
+TODO: Finish describing this
+* stop-requests can be issued concurrently from another thread 
+* usually need to either start the operation or attach a stop-callback
+  * if stop-callback runs before launching the op then may need synchronisation to avoid cancelling the operation before it's started.
+  * if starting the operation befor launching then may need synchronisation to signal when the stop-callback has been subsequently registered so that if the operation completes on another thread it doesn't try to deregister the stop-callback before it's finished registering
+* For example of how this is solved see following async_read() example.
+
+
+### Cancellation Example: Socket async_read() ###
```

Comment:
This example is a big blob of code, and while I think it is helpful, I wonder if it should go into an appendix instead of being right here inline with the rest of the text.

Reply:
Yes, that was my intent - to move this to an appendix.

Although it's possible I could trim the example down to some abstract hypothetical OS call.
e.g. `void begin_os_call(int fd, void* buf, size_t len, void(*cb)(std::error_code, size_t, void*), void* data)`
and `void cancel_os_call(int fd, void* data)`, which would make it easier to explain the essentials of the synchronisation needed to coordinate cancellation.